fixup todo styling

bhavyaus opened the pull request:
- Removed todo widget rendering on top of the chat panel
- Added todo widget rendering on top of the chat input as the only supported UX
- Added a setting to turn off widget display with `chat.tools.todos.showWidget`
- Matched styling of the todos with the edit-session widget

bhavyaus committed:
Update todo styling

bhavyaus committed:
Hide todo list container initially and manage its visibility during rendering

bpasero commented:
There also seems to be an issue when the todo list is empty but the container still applies its styling:

<img width="637" height="211" alt="image" src="https://github.com/user-attachments/assets/b0cff42a-364c-46c9-b830-e778f952ba3b" />

bhavyaus committed:
Remove standalone ChatTodoListWidget and integrate todo UI into ChatInputPart

- Remove ChatTodoListWidget import, field, instantiation and DOM append
- Delegate render/clear of todo list to ChatInputPart (use inputPart.renderChatTodoListWidget / inputPart.clearTodoListWidget)
- Update content height and layout math to exclude removed standalone widget height
- Simplify and relocate todo-list CSS: reduce padding/margins, move widget container styling under .interactive-input-part, and remove the large global .chat-todo-list-widget rules

bhavyaus committed:
Account for integrated todo list in chat input layout

- Include todoListWidgetContainerHeight in ChatInputPart contentHeight and layout calculations so the embedded todo UI is reserved in layout
- Rename ChatWidget.renderChatTodoListWidget to renderChatInputTodoListWidget and update the call site
- Increase bottom padding of .chat-editing-session-container to make space for the todo list

bhavyaus committed:
update config

bhavyaus committed:
Update icon size

bhavyaus committed:
Remove standalone Chat Todo List Widget styles from chat.css

bhavyaus committed:
Add onDidUpdateTodos to mock IChatTodoListService in inline chat tests
